routes/api/authRoutes.js

```diff
@@ -0,0 +1,150 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const User = require("../../models/user");
+const authenticateToken = require("../../middleware/authenticateToken");
+
+const {
+  userSchema,
+  subscriptionUpdateSchema,
+} = require("../../validation/userSchemas");
+
+require("dotenv").config();
+
+router.post("/register", async (req, res) => {
+  try {
+    const { error, value } = userSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+
+    const existingUser = await User.findOne({ email: value.email });
+    if (existingUser) {
+      return res.status(409).json({ message: "Email in use" });
+    }
+
+    const hashedPassword = await bcrypt.hash(value.password, 10);
+
+    const newUser = new User({
+      email: value.email,
+      password: hashedPassword,
+      subscription: "starter",
+    });
+    await newUser.save();
+
+    res.status(201).json({
+      user: {
+        email: newUser.email,
+        subscription: newUser.subscription,
+      },
+    });
+  } catch (error) {
+    console.error("Registration error:", error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+});
+
+router.post("/login", async (req, res) => {
+  try {
+    const { error, value } = userSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+
+    const user = await User.findOne({ email: value.email });
+    if (!user) {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+
+    const passwordMatch = await bcrypt.compare(value.password, user.password);
+    if (!passwordMatch) {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+
+    const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET);
+
+    user.token = token;
+    await user.save();
+
+    res.status(200).json({
+      token,
+      user: {
+        email: user.email,
+        subscription: user.subscription,
+      },
+    });
+  } catch (error) {
+    console.error("Login error:", error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+});
+
+router.post("/users/logout", authenticateToken, async (req, res) => {
+  try {
+    const userId = req.user._id;
+
+    const user = await User.findById(userId);
```

Comment:
W authenticateToken nastąpiła autoryzacja. Więc jeśli użytkownik jest tutaj to już musi być autoryzowany. Po prostu zrób findandupdate albo zobacz czy nie możesz pracować na 
`const user = req.user`
`user.token = null`
`user.save()`